client/src/lib/tests/attempt-result.test.js

```diff
@@ -619,3 +620,34 @@ describe("incompleteMean", () => {
     expect(incompleteMean(attemptResults, "333fm")).toEqual(2200);
   });
 });
+
+describe("computeProjectedAverage", () => {
+  it("Returns average if it is populated", () => {
+    expect(computeProjectedAverage({average: 100}, null)).toEqual(100);
+  });
+
+  var result;
+  it("Returns mean when format is mean of 3", () => {
+    const format = { numberOfAttempts: 3};
+    result = {attempts: [{result: 100}]};
+    expect(computeProjectedAverage(result, format)).toEqual(100);
+    result = {attempts: [{result: 100}, {result: 102}]};
+    expect(computeProjectedAverage(result, format)).toEqual(101);
```

Comment:
Let's test that 0 and 3 attempts also behave as expected (basically we want to test every possible length).